rust-lang/cargo
# Remove `src/doc` from `exclude` list in Cargo.toml

### What does this PR try to resolve?

To remove `src/doc` from `exclude` list in Cargo.toml, we need to do two things

* Make `src/doc/semver-check` a workspace member. We achieve this by moving it to `crates/semver-check`.
* Run `linkchecker.sh` from target directory, so that we only need to exclude `target` from workspace members.

### How should we test and review this PR?

Run two CI job locally. For linkchecker,

```bash
cd target
curl -sSLO https://raw.githubusercontent.com/weihanglo/rust/linkchecker/src/tools/linkchecker/linkcheck.sh
sh linkcheck.sh --all --path ../src/doc cargo
```

For semver-check, just run `cargo +stable run -p semver-check`.

### Additional information

Blocked on https://github.com/rust-lang/rust/pull/110565
